vscode/src/streamingRunner.ts

```diff
@@ -272,6 +296,11 @@ export class StreamingRunner implements vscode.Disposable {
       this.connection.dispose();
     }
 
+    if (this.run!.name === "on_demand_run_in_terminal") {
+      this.run!.end();
+    }
+    this.run = undefined;
```

Comment:
Is it possible that `this.run!` is already `undefined` when this runs the first time?